lib/cog/command/pipeline/executor.ex

```diff
@@ -653,6 +655,31 @@ defmodule Cog.Command.Pipeline.Executor do
     end)
   end
 
+  defp assign_relay(relays, bundle_name, bundle_version) do
+    case Map.get(relays, bundle_name) do
+      # No assignment so let's pick one
+      nil ->
+        case Relays.pick_one(bundle_name, bundle_version) do
+          # No relays available
+          nil ->
+            {nil, relays}
+          # Store the selected relay in the relay cache
+          relay ->
+            {relay, Map.put(relays, bundle_name, relay)}
+        end
+      # Relay was previously assigned
+      relay ->
+        # Is the bundle still available on the relay? If not, remove the current assignment from the cache
+        # and select a new relay
+        if Relays.relay_available?(relay, bundle_name, bundle_version) == false do
```

Comment:
Minor, but I'd probably flip this around to have the `true` arm  come first, simplifying the conditional.